Sources/ApolloAPI/GraphQLNullable.swift

```diff
@@ -1,5 +1,12 @@
 import Foundation
 
+/// Indicates the presence of a value, supporting both `nil` and `null` values.
+///
+/// In GraphQL, explicitly providing a `null` value for an input value to a field argument is
+/// semantically different from not providing a value at all (`nil`). This enum allows you to
+/// distinguish your input values between `null` and `nil`.
```

Comment:
I would add something specific here about how sending a null value generally clears any existing value but not sending a value at all does nothing

Reply:
So, that isn't actually enforced or even specifically stated as part of the spec for how `null` values are used. 
In the GraphQL spec that is linked in this documentation all that is guaranteed is that a `null` value "may be interpreted differently" than a `nil`. It also very often is NOT any different. That's up to your server implementation details.

The spec docs do use clearing an existing value as the primary example of how these things "could" be semantically different, but I don't think that it's appropriate to include that here. I don't want any confusion where people think that **IS** the difference here and they can just pass `null` and assume their server will delete fields. 

I generally really like pointing to the GraphQL spec and letting engineers learn how to actually use GraphQL.

Reply:
@calvincestari @hwillson What are your opinions here? If the team thinks adding that example to the documentation is better, I will.

I generally have a documentation style that is very technical. It is 100% accurate and direct. But it is ONLY accurate and direct to exactly what the code is doing. I try not to make generalizations or assume the reader has any other knowledge or is using my code in any specific use case.

I feel like that kind of commentary is better left to tutorials and usage guides, not the inline-documentation. But I am fine with adding it here if you would like.

Reply:
Great suggestion - I agree that an example might be better suited in our non-API docs, especially in this case where things aren't enforced by the spec.

Reply:
It's tricky if it's not part of the spec, but I feel like it's probably worth giving an example of *why* this needs to exist - otherwise people may be confused as to why we're using this instead of optionals

Reply:
I also added an explanation of this, with the example to the RFC. (I very much expect that large portions of the RFC will be taken whole sale for the docs site for the release.)

Reply:
Examples certainly help to illustrate concepts and since we're dealing with Swift, where nil and optional are concepts readers will already be familiar with, I think I'd err on the side of caution and go with the example here so the reader doesn't infer their own understanding of what it these may mean.